toolkit/tools/internal/kernelversion/kernelversion.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package kernelversion
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/version"
+)
+
+var (
+	// Parses the kernel version from "uname -r" or subdirectories of /lib/modules.
+	//
+	// Examples:
+	//   OS               Version
+	//   Fedora 40        6.11.6-200.fc40.x86_64
+	//   Ubuntu 22.04     6.8.0-48-generic
+	//   Azure Linux 2.0  5.15.153.1-2.cm2
+	//   Azure Linux 3.0  6.6.47.1-1.azl3
+	kernelVersionRegex = regexp.MustCompile(`^(\d+)\.(\d+)\.(\d+)([.\-][a-zA-Z0-9_.\-]*)?$`)
+)
+
+func GetBuildHostKernelVersion() (version.Version, error) {
```

Comment:
You could also use `GetKernelVersion` from this package: https://pkg.go.dev/github.com/moby/moby/pkg/parsers/kernel#GetKernelVersion.

Reply:
This is simple enough that I don't think it is worth taking an external dependency. Though, I did switch to using the `uname` function directly.